lib/python/qmk/makefile.py

```diff
@@ -0,0 +1,72 @@
+""" Functions for working with Makefiles
+"""
+import os
+import glob
+import re
+
+import qmk.path
+from qmk.errors import NoSuchKeyboardError
+
+def parse_rules_mk(file_path):
+    """ Parse a rules.mk file
+
+    Args:
+        file_path: path to the rules.mk file
+
+    Returns:
+        a dictionary with the file's content
+    """
+    # regex to match lines uncommented lines and get the data
+    # group(1) = option's name
+    # group(2) = operator (eg.: '=', '+=')
+    # group(3) = value(s)
+    rules_mk_regex = re.compile(r"^\s*(\w+)\s*([\?\:\+\-]?=)\s*(\S.*?)(?=\s*(\#|$))")
```

Comment:
I strongly prefer to keep regex to a minimum. It's hard to spot bugs in and virtually no one double checks it. Can you rework this to not use a regex?